cellprofiler/gui/moduleview.py

```diff
@@ -4464,18 +4467,21 @@ def __init__(self, pipeline, module, callback):
     def cancel(self):
         self.cancelled = True
 
-
-def cache_pipeline(pipeline):
-    """Return a single cached copy of a pipeline to limit the # of copies"""
-    d = getattr(request_pipeline_cache, "d", None)
-    if d is None:
-        d = weakref.WeakValueDictionary()
-        setattr(request_pipeline_cache, "d", d)
-    settings_hash = pipeline.settings_hash()
-    result = d.get(settings_hash)
-    if result is None:
-        result = d[settings_hash] = pipeline.copy(False)
-    return result
+#
+# Cacheing is not compatible with the current pattern of sharing state between GUI components. E.g., the pipeline object
+# stored in the ModuleView constructor will never be updated, yet it is used to validate current state.
```

Comment:
I decided to leave this as a comment, but I am not sure how useful it will be in the future. This change was critical to solving the problem where the pipeline would be updated but the change would not propagate through the GUI. 

I am not certain how critical pipeline caching is (especially for large pipelines). It is probably worth investigating. However, I think the larger issue is not the size of the pipeline but the number of generated validation requests.